forem/forem
# Don't create podcast episodes without any media

## What type of PR is this? (check all applicable)


## Description

There is an early return from add_media_url to prevent using an empty
enclosure_url, which avoids errors in
`GetMediaUrl.call(a_blank_value)`, however there's also a not null
constraint on podcast_episodes.media_url in the database, which causes
upsert to raise an error.

Returning early here avoids the error (which had halted the worker).

See also
https://app.honeybadger.io/projects/66984/faults/80833402

This error happens regularly, 96 times per day in DEV, it looks like at least
one podcast feed_url is for an rss feed filtered by /tag/podcast/,
(and at least one post was tagged "podcast" but didn't include any
media to get).

My understanding is at worst we'll ignore the error silently.



## Related Tickets & Documents

https://app.honeybadger.io/projects/66984/faults/80833402

## QA Instructions, Screenshots, Recordings

The issue in DEV is related to the rss feed at https://codepunk.io/podcast/rss/codepunk.xml - if you create a podcast with this feed url, and import episodes, no sidekiq failures should be observed.



### UI accessibility concerns?

None

## Added/updated tests?

- [ ] No, and this is why: ~feeling lazy.... okay okay I'll write a test!~

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._


## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](gif_link)
